build/azure-pipelines/cli/install-rust.yml

```diff
@@ -0,0 +1,46 @@
+parameters:
+  - name: channel
+    type: string
+    default: stable
+  - name: targets
+    default: []
+    type: object
+
+# Todo: use 1ES pipeline once extension is installed in ADO
+
+steps:
+  - script: |
+      set -e
+      curl https://sh.rustup.rs -sSf | sh -s -- -y --profile minimal --default-toolchain $RUSTUP_TOOLCHAIN
+      echo "##vso[task.setvariable variable=PATH;]$PATH:$HOME/.cargo/bin"
+    env:
+      RUSTUP_TOOLCHAIN: ${{ parameters.channel }}
+    condition: not(eq(variables['Agent.OS'], 'Windows_NT'))
```

Comment:
We survived so far by having platform specific pipelines for stuff like this. This PR introduces two new ways to distinguish between Posix and Win32. The first I've seen was `VSCODE_IS_POSIX`. Now I see this. Please simplify this and just have platform specific pipelines.